src/DotNetWorker.Core/Context/FunctionContextBindingFeatureExtensions.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Context.Features;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// FunctionContext extension methods for binding data.
+    /// </summary>
+    public static class FunctionContextBindingFeatureExtensions
+    {
+        /// <summary>
+        /// Binds an input binding item for the requested <see cref="BindingMetadata"/> instance.
+        /// </summary>
+        /// <param name="context">The function context.</param>
+        /// <param name="bindingMetadata">The BindingMetadata instance for which input data should bound to.</param>
+        /// <returns>An instance of T if binding was successful, else null</returns>
+        public static async ValueTask<T?> BindInputAsync<T>(this FunctionContext context, BindingMetadata bindingMetadata)
+        {
```

Comment:
Didn't we talk at some point about this returning a wrapper type like `BindingResult<T>`, which would let you completely replace the value in the cache if you wanted to?

I may be misremembering, though :)

Reply:
I see we have similar concepts for `InvocationResult` and `OutputBindingData`...

Reply:
You are remembering it correctly. We did discuss it. In the latest iteration, I wrapped the result in `InputBindingData<T>` type. 
One thing we may consider doing is replacing the new cache with `IFunctionBindingsFeature` (that is what the other 2 items (InvocationResult and OutputBindingData is relying on to update the value). Will follow up offline to discuss this.